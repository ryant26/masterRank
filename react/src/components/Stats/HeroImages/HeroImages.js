import React from 'react';
import Proptypes from 'prop-types';

import DisableableHeroImage from '../../Images/DisableableHeroImage/DisableableHeroImage';
import HeroImage from '../../Images/HeroImage/HeroImage';

const HeroImages = ({heroNames, disabled}) => {

    let heroImages = heroNames.map((name, i) => {
        return ( disabled[i]
            ? ( <DisableableHeroImage
                    key={i}
                    heroName={name}
                    disabled={true}
                />)
            : ( <HeroImage
                 key={i}
                 heroName={name}
                />)
        );
    });

    return (
        <div className="HeroImages flex">
            { heroImages }
        </div>
    );
};

HeroImages.defaultProps = {
    disabled: []
};

HeroImages.propTypes = {
    heroNames: Proptypes.array.isRequired,
    disabled: Proptypes.arrayOf(Proptypes.bool)
};

export default HeroImages;