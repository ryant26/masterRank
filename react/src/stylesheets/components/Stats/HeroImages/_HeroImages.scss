.HeroImages {

  .HeroImage {
    @include square;
    margin-right: 10px;
  }

  .disabled {
    @include square;
    margin-right: 10px;
  }
}