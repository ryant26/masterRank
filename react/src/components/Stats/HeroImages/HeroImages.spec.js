import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import HeroImages from './HeroImages';

const names = require('../../../../../shared/allHeroNames').names;

describe('HeroCard Component',()=> {
    it('should render without exploding', () => {
        const wrapper = mount(
            <HeroImages heroNames={[names[0], names[1], names[2]]}/>
        );

        const HeroCardComponent = wrapper.find(HeroImages);
        expect(HeroCardComponent).toBeTruthy();
    });

    it('should match the snapshot', () => {
        let component = renderer.create(
            <HeroImages heroNames={[names[0], names[1], names[2]]}/>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});