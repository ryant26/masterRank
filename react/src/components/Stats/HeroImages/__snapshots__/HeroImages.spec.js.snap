// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCard Component should match the snapshot 1`] = `
<div
  className="HeroImages flex"
>
  <img
    alt="doomfist icon"
    className="HeroImage"
    onClick={undefined}
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <img
    alt="genji icon"
    className="HeroImage"
    onClick={undefined}
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <img
    alt="mccree icon"
    className="HeroImage"
    onClick={undefined}
    src={
      Object {
        "process": [Function],
      }
    }
  />
</div>
`;
